<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>680</width>
    <height>500</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QGroupBox" name="GroupControls">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Controls</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="7" column="0" colspan="2">
          <widget class="QGroupBox" name="groupBox">
           <property name="title">
            <string>Projection Mode</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QRadioButton" name="projection_Radio_Persp">
              <property name="text">
               <string>Perspective</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="projection_Radio_Orth">
              <property name="text">
               <string>Orthogonal</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QSpinBox" name="angle_Spinbox">
           <property name="maximum">
            <number>360</number>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QSpinBox" name="fov_Spinbox">
           <property name="minimum">
            <number>45</number>
           </property>
           <property name="maximum">
            <number>180</number>
           </property>
          </widget>
         </item>
         <item row="11" column="0" colspan="2">
          <widget class="QPushButton" name="pushButton">
           <property name="text">
            <string>Reset</string>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QSlider" name="angle_Slider">
           <property name="maximum">
            <number>360</number>
           </property>
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
          </widget>
         </item>
         <item row="8" column="0" colspan="2">
          <widget class="QGroupBox" name="groupBox_2">
           <property name="title">
            <string>Rotations</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_3">
            <item row="4" column="0">
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>Ring C</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Ring A</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QSlider" name="rotation_B_Slider">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QSlider" name="rotation_C_Slider">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSlider" name="rotation_A_Slider">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>Ring B</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="10" column="0" colspan="2">
          <widget class="QGroupBox" name="groupBox_3">
           <property name="title">
            <string>Clipping Plane</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="1" column="0">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>Near</string>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Far</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QDoubleSpinBox" name="near_Spinbox">
              <property name="value">
               <double>0.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QDoubleSpinBox" name="far_SpinBox">
              <property name="value">
               <double>2.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="FieldOfView_Label">
           <property name="text">
            <string>Field of view</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLabel" name="Angle_Label">
           <property name="text">
            <string>Angle</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="5" column="0" rowspan="2">
          <widget class="QSlider" name="fov_Slider">
           <property name="minimum">
            <number>45</number>
           </property>
           <property name="maximum">
            <number>180</number>
           </property>
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="MyGLWidget" name="myOpenGLWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>480</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>myglwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>angle_Spinbox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>angle_Slider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>122</x>
     <y>71</y>
    </hint>
    <hint type="destinationlabel">
     <x>107</x>
     <y>159</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>angle_Slider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>angle_Spinbox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>107</x>
     <y>168</y>
    </hint>
    <hint type="destinationlabel">
     <x>122</x>
     <y>71</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fov_Spinbox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>fov_Slider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>79</x>
     <y>71</y>
    </hint>
    <hint type="destinationlabel">
     <x>41</x>
     <y>159</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fov_Slider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>fov_Spinbox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>41</x>
     <y>168</y>
    </hint>
    <hint type="destinationlabel">
     <x>79</x>
     <y>71</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
